opt_einsum/backends/array_api.py

```diff
@@ -0,0 +1,75 @@
+"""
+Required functions for optimized contractions of arrays using array API-compliant backends.
+"""
+import sys
+from typing import Callable
+from types import ModuleType
+
+import numpy as np
+
+from ..sharing import to_backend_cache_wrap
+
+
+def discover_array_apis():
+    """Discover array API backends."""
+    if sys.version_info >= (3, 8):
+        from importlib.metadata import entry_points
+
+        if sys.version_info >= (3, 10):
+            eps = entry_points(group="array_api")
+        else:
+            # Deprecated - will raise warning in Python versions >= 3.10
+            eps = entry_points().get("array_api", [])
+        return [ep.load() for ep in eps]
+    else:
+        # importlib.metadata was introduced in Python 3.8, so it isn't available here. Unable to discover any array APIs.
+        return []
```

Comment:
Numpy does not officially support below 3.8: https://numpy.org/neps/nep-0029-deprecation_policy.html

It would be worth considering dropping Python 3.7 as well. @jcmgray what do you think?